### PR TITLE
*: only use go.uber.org/atomic as atomic dependency

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

remove `github.com/siddontang/go`

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -125,7 +124,7 @@ func (w *Worker) Close() {
 		failpoint.Return()
 	})
 
-	if !atomic.CompareAndSwapInt64(&w.closed, 0, 1) {
+	if w.closed.CAS(false, true) {
```
Comment (lance6716) [inline loader/loader.go:127]:
```suggestion
	if !w.closed.CAS(false, true) {
```

**Comment by lance6716** [comment]
ptal @glorv @Ehco1996

File: dm/master/workerrpc/rawgrpc.go [hunk 2]
```diff
@@ -31,15 +31,15 @@ import (
 type GRPCClient struct {
 	conn   *grpc.ClientConn
 	client pb.WorkerClient
-	closed int32
+	closed *atomic.Bool
```
Comment (glorv) [inline dm/master/workerrpc/rawgrpc.go:34]:
```suggestion
	closed atomic.Bool
```

Do not use pointer to be consistance with other declares.

File: pkg/utils/file.go [hunk 3]
```diff
@@ -64,3 +67,24 @@ func GetFileSize(file string) (int64, error) {
...
+	}
+	n, err := f.Write(data)
+	f.Close()
+	if err == nil && n < len(data) {
+		err = io.ErrShortWrite
+	} else {
+		err = os.Chmod(f.Name(), perm)
+	}
+	if err != nil {
+		os.Remove(f.Name())
```
Comment (glorv) [inline pkg/utils/file.go:86]: what if `os.Remove` call fails here, do we need to output a log?

**Review by glorv — COMMENTED** [review]
rest LGTM

File: pkg/utils/file.go (hunk 3 above)
Comment (lance6716) [reply pkg/utils/file.go]: https://github.com/pingcap/dm/pull/1615/commits/feab697c8a39b13e80ca8eaac0a0b3c8265105a7

**Pull request merged**